src/index.js

```diff
@@ -37,7 +37,8 @@ function buildValidations(referenced, dereferenced, receivedOptions) {
     const schemas = {};
 
     const basePaths = dereferenced.servers && dereferenced.servers.length
-        ? dereferenced.servers.map(({ url }) => new URL(url).pathname)
+        // dummy base path is required by URL constructor when server url is not absolute
+        ? dereferenced.servers.map(({ url }) => new URL(url, 'http://example.com').pathname)
```

Comment:
Would matching wirk correctly with this hardcoded value? Definitely needs a new e2e test for this case of matching.